<?xml version="1.0" encoding="utf-8" ?>

<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot name="urdf_new">
    <link name="world" />
    <joint name="world_base_joint" type="fixed">
        <parent link="world" />
        <child link="Link_1" />
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.01" />
    </joint>
    <link name="Link_1">
        <inertial>
            <origin xyz="0.00054038 0.00054918 0.14441" rpy="0 0 0" />
            <mass value="0.9506" />
            <inertia ixx="0.001726" ixy="2.7336E-05" ixz="-1.1307E-07" iyy="0.0012391" iyz="7.2951E-08" izz="0.0028405" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_1.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual> 
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_1.STL" />
            </geometry>
        </collision>
    </link>
    <link name="Link_2">
        <inertial>
            <origin xyz="-7.8775E-05 0.006832 0.11498" rpy="0 0 0" />
            <mass value="0.8104" />
            <inertia ixx="0.0006824" ixy="-4.3606E-06" ixz="1.0839E-08" iyy="0.001058" iyz="-1.4562E-07" izz="0.00076619" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_2.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_2.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="Joint_1" type="continuous">
        <origin xyz="0 0 0.1305" rpy="0 0 0" />
        <parent link="Link_1" />
        <child link="Link_2" />
        <axis xyz="0 0 1" />
    </joint>
    <link name="Link_3">
        <inertial>
            <origin xyz="-0.011418 -0.078842 0.26809" rpy="0 0 0" />
            <mass value="0.85091" />
            <inertia ixx="0.0031641" ixy="-5.4061E-06" ixz="0.00013238" iyy="0.0036446" iyz="1.5693E-06" izz="0.00074596" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_3.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_3.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="Joint_2" type="continuous">
        <origin xyz="0 0 0.154" rpy="0 0 0" />
        <parent link="Link_2" />
        <child link="Link_3" />
        <axis xyz="-0.011543 0.99993 0" />
    </joint>
    <link name="Link_4">
        <inertial>
            <origin xyz="-0.018707 -0.024132 0.18955" rpy="0 0 0" />
            <mass value="0.50318" />
            <inertia ixx="0.004614" ixy="-1.9766E-06" ixz="0.00043139" iyy="0.0047852" iyz="4.98E-06" izz="0.00034952" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_4.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_4.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="Joint_3" type="continuous">
        <origin xyz="-0.018642 -0.0002152 0.38856" rpy="0 0 0" />
        <parent link="Link_3" />
        <child link="Link_4" />
        <axis xyz="0.011543 -0.99993 0" />
    </joint>
    <link name="Link_5">
        <inertial>
            <origin xyz="-0.010421 -0.00012033 0.10406" rpy="0 0 0" />
            <mass value="0.37501" />
            <inertia ixx="0.00035594" ixy="2.2825E-07" ixz="1.1743E-05" iyy="0.00036198" iyz="1.6536E-07" izz="0.00023987" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_5.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_5.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="Joint_4" type="continuous">
        <origin xyz="-0.041155 -0.015476 0.41268" rpy="0 0 0" />
        <parent link="Link_4" />
        <child link="Link_5" />
        <axis xyz="0.099647 0.0011503 -0.99502" />
    </joint>
    <link name="Link_6">
        <inertial>
            <origin xyz="-0.00083373 4.5557E-06 0.006" rpy="0 0 0" />
            <mass value="0.41858" />
            <inertia ixx="0.0002151" ixy="5.6507E-06" ixz="2.4383E-07" iyy="0.00032873" iyz="6.9637E-07" izz="0.00021335" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_6.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_6.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="Joint_5" type="continuous">
        <origin xyz="-0.020647 -0.00011557 0.18613" rpy="0 0 0" />
        <parent link="Link_5" />
        <child link="Link_6" />
        <axis xyz="0.049557 0.99875 0.0061175" />
    </joint>
    <link name="Link_7">
        <inertial>
            <origin xyz="-0.03465 0.011342 0.18658" rpy="0 0 0" />
            <mass value="0.60212" />
            <inertia ixx="0.00046695" ixy="8.2195E-05" ixz="4.3056E-05" iyy="0.00051681" iyz="2.9598E-06" izz="0.0002398" />
        </inertial>
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_7.STL" />
            </geometry>
            <material name="">
                <color rgba="0.79216 0.81961 0.93333 1" />
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <mesh filename="package://urdf_new/meshes/Link_7.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="Joint_7" type="continuous">
        <origin xyz="0.0098543 -0.00047897 -0.001632" rpy="0 0 0" />
        <parent link="Link_6" />
        <child link="Link_7" />
        <axis xyz="-0.13763 0.00076231 0.99048" />
    </joint>
    <transmission name="trans_Joint_1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="Joint_1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="Joint_1_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_Joint_2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="Joint_2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="Joint_2_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_Joint_3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="Joint_3">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="Joint_3_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_Joint_4">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="Joint_4">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="Joint_4_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_Joint_5">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="Joint_5">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="Joint_5_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_Joint_7">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="Joint_7">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="Joint_7_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <!-- <gazebo>
        <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
            <robotNamespace>/</robotNamespace>
        </plugin>
    </gazebo> -->

    <!-- <ros2_control name="GazeboSystem" type="system">
  <hardware>
    <plugin>gazebo_ros2_control/GazeboSystem</plugin>
  </hardware>
</ros2_control> -->

<gazebo>
  <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
  </plugin>
</gazebo>

<joint name="camera_joint" type="fixed">
  <parent link="camera_link" />
  <child link="camera_bottom_screw_frame" />
  <origin xyz="0 0.0 0" rpy="0 0 0" />
</joint>


<link name="camera_link_depth"></link>
<link name="camera_bottom_screw_frame"></link>
<link name="camera_depth_frame"></link>
<link name="camera_depth_optical_frame"></link>
<link name="camera_color_frame"></link>
<link name="camera_color_optical_frame"></link>
<link name="camera_left_ir_frame"></link>
<link name="camera_left_ir_optical_frame"></link>
<link name="camera_right_ir_frame"></link>
<link name="camera_right_ir_optical_frame"></link>

<joint name="camera_link_joint" type="fixed">
  <parent link="camera_bottom_screw_frame" />
  <child link="camera_link_depth" />
  <origin xyz="0 0.0 0.0 " rpy="0 0 0" />
</joint>

<joint name="camera_depth_joint" type="fixed">
  <parent link="camera_link_depth" />
  <child link="camera_depth_frame" />
  <origin xyz="0 0 0" rpy="0 0 0" />
</joint>
<joint name="camera_depth_optical_joint" type="fixed">
  <parent link="camera_depth_frame" />
  <child link="camera_depth_optical_frame" />
  <origin xyz="0 0 0 " rpy="-1.57 0 -1.57" />
</joint>
<joint name="camera_color_joint" type="fixed">
  <parent link="camera_depth_frame" />
  <child link="camera_color_frame" />
  <origin xyz="0 0 0" rpy="0 0 0" />
</joint>
<joint name="camera_color_optical_joint" type="fixed">
  <parent link="camera_color_frame" />
  <child link="camera_color_optical_frame" />
  <origin xyz="0 0 0 " rpy="-1.57 0 -1.57" />
</joint>
<joint name="camera_left_ir_joint" type="fixed">
  <parent link="camera_depth_frame" />
  <child link="camera_left_ir_frame" />
  <origin xyz="0 0 0 " rpy="0 0 0 " />
</joint>
<joint name="camera_left_ir_optical_joint" type="fixed">
  <parent link="camera_left_ir_frame" />
  <child link="camera_left_ir_optical_frame" />
  <origin xyz="0 0 0 " rpy="-1.57 0 -1.57" />
</joint>
<joint name="camera_right_ir_joint" type="fixed">
  <parent link="camera_depth_frame" />
  <child link="camera_right_ir_frame" />
  <origin xyz="0 -0.050 0 " rpy="0 0 0" />
</joint>
<joint name="camera_right_ir_optical_joint" type="fixed">
  <parent link="camera_right_ir_frame" />
  <child link="camera_right_ir_optical_frame" />
  <origin xyz="0 0 0 " rpy="-1.57 0 -1.57" />
</joint>



  <gazebo reference="camera_bottom_screw_frame"> 
        <sensor name="cameradepth" type="depth">
                <camera name="camera">
                    <horizontal_fov>1.2043</horizontal_fov>
                    <image>
                        <width>1280</width>
                        <height>720</height>
                        <format>R8G8B8</format>
                    </image>
                    <clip>
                        <near>0.1</near>
                        <far>100</far>
                    </clip>
                </camera>
                <always_on>1</always_on>
                <update_rate>30</update_rate>
                <visualize>0</visualize>
            </sensor>
            <sensor name="cameracolor" type="camera">
                <camera name="camera">
                    <horizontal_fov>1.2043</horizontal_fov>
                    <image>
                        <width>1280</width>
                        <height>720</height>
                        <format>RGB_INT8</format>
                    </image>
                    <clip>
                        <near>0.1</near>
                        <far>100</far>
                    </clip>
                    <noise>
                        <type>gaussian</type>
                        <mean>0.0</mean>
                        <stddev>0.007</stddev>
                    </noise>
                </camera>
                <always_on>1</always_on>
                <update_rate>30</update_rate>
                <visualize>0</visualize>
            </sensor>
            <sensor name="cameraired1" type="camera">
                <camera name="camera">
                    <horizontal_fov>1.2043</horizontal_fov>
                    <image>
                        <width>1280</width>
                        <height>720</height>
                        <format>L_INT8</format>
                    </image>
                    <clip>
                        <near>0.1</near>
                        <far>100</far>
                    </clip>
                    <noise>
                        <type>gaussian</type>
                        <mean>0.0</mean>
                        <stddev>0.05</stddev>
                    </noise>
                </camera>
                <always_on>1</always_on>
                <update_rate>1</update_rate>
                <visualize>0</visualize>
            </sensor>
            <sensor name="cameraired2" type="camera">
                <camera name="camera">
                    <horizontal_fov>1.2043</horizontal_fov>
                    <image>
                        <width>1280</width>
                        <height>720</height>
                        <format>L_INT8</format>
                    </image>
                    <clip>
                        <near>0.1</near>
                        <far>100</far>
                    </clip>
                    <noise>
                        <type>gaussian</type>
                        <mean>0.0</mean>
                        <stddev>0.05</stddev>
                    </noise>
                </camera>
                <always_on>1</always_on>
                <update_rate>1</update_rate>
                <visualize>0</visualize>
            </sensor>     
  </gazebo>

  <gazebo>
  <plugin name="camera" filename="librealsense_gazebo_plugin.so">
        <prefix>camera</prefix>
        <depthUpdateRate>15.0</depthUpdateRate>
        <colorUpdateRate>15.0</colorUpdateRate>
        <infraredUpdateRate>15.0</infraredUpdateRate>
        <depthTopicName>camera/aligned_depth_to_color/image_raw</depthTopicName>
        <depthCameraInfoTopicName>camera/depth/camera_info</depthCameraInfoTopicName>
        <colorTopicName>camera/color/image_raw</colorTopicName>
        <colorCameraInfoTopicName>camera/color/camera_info</colorCameraInfoTopicName>
        <infrared1TopicName>camera/infra1/image_raw</infrared1TopicName>
        <infrared1CameraInfoTopicName>camera/infra1/camera_info</infrared1CameraInfoTopicName>
        <infrared2TopicName>camera/infra2/image_raw</infrared2TopicName>
        <infrared2CameraInfoTopicName>camera/infra2/camera_info</infrared2CameraInfoTopicName>
        <colorOpticalframeName>camera_color_optical_frame</colorOpticalframeName>
        <depthOpticalframeName>camera_depth_optical_frame</depthOpticalframeName>
        <infrared1OpticalframeName>camera_left_ir_optical_frame</infrared1OpticalframeName>
        <infrared2OpticalframeName>camera_right_ir_optical_frame</infrared2OpticalframeName>
        <rangeMinDepth>0.1</rangeMinDepth>
        <rangeMaxDepth>2.0</rangeMaxDepth>
        <pointCloud>true</pointCloud>
        <pointCloudTopicName>camera/depth/color/points</pointCloudTopicName>
        <pointCloudCutoff>0.1</pointCloudCutoff>
        <pointCloudCutoffMax>2.0</pointCloudCutoffMax>
  </plugin>

  </gazebo>
    <joint name="realsense_joint" type="fixed">
    <parent link="world" />
    <child link="camera_link" />
    <origin xyz="+0.2 -0.1 0.5" rpy="0.0 0.261 3.104" /> 
  </joint>


  <link name="camera_link">  
  <collision>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <geometry>
      <box size="0.001 0.001 0.001"/>
    </geometry>
  </collision>

  <visual>
    <origin xyz="0 0 0" rpy="1.57 0 1.57"/>  
    <geometry>
        <mesh filename="file:///home/agrim/eyrc-24-25-logistic-cobot/ur_description/meshes/NAVRS.STL" />
    </geometry>
  </visual>

  <inertial>
    <mass value="1e-5" />
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0" />
  </inertial>
  </link>
</robot>
